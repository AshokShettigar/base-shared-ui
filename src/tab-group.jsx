import classNames from 'classnames';
import { cloneElement } from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line import/prefer-default-export
export const TabGroup = props => {
    const className = classNames('cn-tabs', props.className);
    return (
        <div>
            <ul className={className}>
                {props.tabs.map(tab => cloneElement(tab, { key: tab.props.label }))}
            </ul>
            {props.children &&
                <div className="cn-tabs__content">
                    {props.children}
                </div>
            }
        </div>
    );
};

TabGroup.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    tabs: PropTypes.node.isRequired
};
