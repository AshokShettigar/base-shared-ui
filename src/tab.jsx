/* eslint-disable react/no-unused-prop-types */
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export function HigherOrderTabComponent(ComposedComponent) {
    const HOCTab = props => {
        const tabClassName = classNames({
            'cn-tab': true,
            'cn-tab--active': props.isActive
        }, props.tabClassName);
        const className = classNames('cn-tab__link', props.className);
        return (
            <li className={tabClassName}>
                <ComposedComponent {...props} className={className} />
            </li>
        );
    };

    HOCTab.propTypes = {
        className: PropTypes.string,
        isActive: PropTypes.bool.isRequired,
        label: PropTypes.string.isRequired,
        tabClassName: PropTypes.string
    };

    HOCTab.defaultProps = {
        isActive: false
    };

    return HOCTab;
}

export const DefaultTab = HigherOrderTabComponent(
    props => (
        <Link to={props.href} className={props.className}>
            {props.label}
        </Link>
        )
    );

DefaultTab.propTypes = {
    ...DefaultTab.propTypes,
    href: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.shape({
            pathname: PropTypes.string,
            search: PropTypes.string,
            hash: PropTypes.string
        })
    ]).isRequired
};
