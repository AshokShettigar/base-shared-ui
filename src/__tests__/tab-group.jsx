import { shallow } from 'enzyme';
import { DefaultTab as Tab } from 'platform-common-ui/tab';
import { TabGroup } from 'platform-common-ui/tab-group';

describe('tab-group', function() {
    const tabs = [
        <Tab href="#" label="tab1" />,
        <Tab href="#" label="tab2" />
    ];

    it('should render the tab group', function() {
        const tree = shallow(
            <TabGroup tabs={tabs} />
        );
        expect(tree).toMatchSnapshot();
    });

    it('should render the tab group with class name', function() {
        const tree = shallow(
            <TabGroup className="my-class" tabs={tabs} />
        );
        expect(tree).toMatchSnapshot();
    });

    it('should render the tab group with content', function() {
        const tree = shallow(
            <TabGroup tabs={tabs}>
                Some content
            </TabGroup>
        );
        expect(tree).toMatchSnapshot();
    });
});
