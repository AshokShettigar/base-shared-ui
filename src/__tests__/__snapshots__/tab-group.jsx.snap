// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tab-group should render the tab group 1`] = `
<div
  data-component="TabGroup"
>
  <ul
    className="cn-tabs"
  >
    <HOCTab
      href="#"
      isActive={false}
      label="tab1"
    />
    <HOCTab
      href="#"
      isActive={false}
      label="tab2"
    />
  </ul>
</div>
`;

exports[`tab-group should render the tab group with class name 1`] = `
<div
  data-component="TabGroup"
>
  <ul
    className="cn-tabs my-class"
  >
    <HOCTab
      href="#"
      isActive={false}
      label="tab1"
    />
    <HOCTab
      href="#"
      isActive={false}
      label="tab2"
    />
  </ul>
</div>
`;

exports[`tab-group should render the tab group with content 1`] = `
<div
  data-component="TabGroup"
>
  <ul
    className="cn-tabs"
  >
    <HOCTab
      href="#"
      isActive={false}
      label="tab1"
    />
    <HOCTab
      href="#"
      isActive={false}
      label="tab2"
    />
  </ul>
  <div
    className="cn-tabs__content"
  >
    Some content
  </div>
</div>
`;
